packages/gravity/proto-guard/src/check-snapshot.test.ts

```diff
@@ -7,90 +7,69 @@ import fs from 'node:fs';
+import { copyDirSync, createConfig, getBaseDataDir } from './util';
 
-describe('Tests against old storage', () => {
-  let testStoragePath: string;
+describe('Tests against storage', () => {
+  const baseDir = getBaseDataDir();
 
-  beforeAll(() => {
-    // Copy storage image to tmp folder against which tests will be run.
-    log.info(`Storage version ${STORAGE_VERSION}`);
+  const copySnapshotToTmp = (snapshot: SnapshotDescription) => {
+    const testStoragePath = path.join('tmp', `proto-guard-${PublicKey.random().toHex()}`);
 
-    testStoragePath = fs.mkdtempSync(path.join('tmp', 'proto-guard-'));
-    const storagePath = path.join(getStorageDir(), STORAGE_VERSION.toString());
+    const storagePath = path.join(baseDir, snapshot.dataRoot);
     log.info('Copy storage', { src: storagePath, dest: testStoragePath });
+    copyDirSync(storagePath, testStoragePath);
+    afterTest(() => fs.rmSync(testStoragePath, { recursive: true }));
 
-    copySync(storagePath, testStoragePath);
-  });
+    return testStoragePath;
+  };
+
+  test('check if space loads for LevelDb snapshot', async () => {
+    const snapshot = SnapshotsRegistry.getSnapshot('levelDB');
+    invariant(snapshot, 'Snapshot not found');
+    log.info('Testing snapshot', { snapshot });
 
-  afterAll(() => {
-    fs.rmSync(testStoragePath, { recursive: true, force: true });
+    const spacesDump = JSON.parse(fs.readFileSync(path.join(baseDir, snapshot.jsonDataPath), 'utf-8'));
+
+    const tmp = copySnapshotToTmp(snapshot);
+    const builder = new TestBuilder(createConfig({ dataRoot: tmp }));
+    afterTest(() => builder.destroy());
+    const services = builder.createLocalClientServices();
+
+    const client = new Client({ services });
+    await asyncTimeout(client.initialize(), 1_000);
+    await client.spaces.isReady.wait();
+    afterTest(() => services.close());
+    afterTest(() => client.destroy());
+
+    expect(await checkIfSpacesMatchDump(client, spacesDump)).to.be.true;
   });
 
-  test('check if space loads', async () => {
-    const builder = new TestBuilder(getConfig(testStoragePath));
+  test('check if space loads for Automerge on nodeFS snapshot', async () => {
+    const snapshot = SnapshotsRegistry.getSnapshot('automerge');
+    invariant(snapshot, 'Snapshot not found');
+    log.info('Testing snapshot', { snapshot });
+
+    const spacesDump = JSON.parse(fs.readFileSync(path.join(baseDir, snapshot.jsonDataPath), 'utf-8'));
+
+    const tmp = copySnapshotToTmp(snapshot);
+    const builder = new TestBuilder(createConfig({ dataRoot: tmp }));
     afterTest(() => builder.destroy());
     const services = builder.createLocalClientServices();
 
-    log.info('running', { storage: services.host?.config?.values.runtime?.client?.storage?.dataRoot });
     const client = new Client({ services });
     await asyncTimeout(client.initialize(), 1_000);
+    await client.spaces.isReady.wait();
     afterTest(() => services.close());
```

Comment:
pair those up with the lines that open those resources:

const services = ...
afterTest(() => services.close())